llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -2311,6 +2322,31 @@ NVPTXTargetLowering::LowerCONCAT_VECTORS(SDValue Op, SelectionDAG &DAG) const {
   return DAG.getBuildVector(Node->getValueType(0), dl, Ops);
 }
 
+SDValue NVPTXTargetLowering::LowerBITCAST(SDValue Op, SelectionDAG &DAG) const {
+  // Handle bitcasting from v2i8 without hitting the default promotion
+  // strategy which goes through stack memory.
+  SDLoc DL(Op);
+
+  EVT ToVT = Op->getValueType(0);
+  EVT FromVT = Op->getOperand(0)->getValueType(0);
+
+  if (FromVT == MVT::v2i8) {
```

Comment:
Prefer early returns: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code